### PR TITLE
Enforce strict_max_version for theme experiments (Fixes #236)

### LINKED ISSUE #236: Force add-ons with theme_experiments to require a strict_max_version
Theme experiments add CSS definitions which apply to core elements, which can change over time. For example, supernova removed the tree for the thread pane and CSS selectors for the tree no longer work.

I therefore suggest to force theme_experiments to include a strict_max_version (during the submission process).

### PR DESCRIPTION
**Pull request by MelissaAutumn**
Fixes #236

Add a property to look for theme experiments, and add it to the submission check for strict_max_version. 

Note: This error message is not localized, so I'm not breaking any existing localization.

### CODE REVIEW

**Comment by jobisoft** [comment]
LGTM. Thanks!

**Comment by tdulcet** [comment]
This change seems to have broken the ATN linter today, as it is now erroneously claiming that my add-on is an experiment when it is actually pure web extension:
image

**Review by tdulcet — COMMENTED** [review]
Can you please revert this commit so that users can upload non-experiment add-ons to ATN again.

File: src/olympia/files/utils.py [hunk 1]
```diff
@@ -387,6 +387,10 @@ def is_experiment(self):
         we just look at the manifest."""
         return bool(self.get('experiment_apis', False))
 
+    def is_theme_experiment(self):
```
[the remark this answers is not in the document]
Comment (MelissaAutumn) [reply src/olympia/files/utils.py:390]: Hello there, thank you for your patience here. We've patched this up, and it's been deployed on ATN. Feel free to open an issue if you experience any other issues uploading your addons.

**Pull request merged**